src/app/shared/example-viewer/example-code-viewer.ts

```diff
@@ -0,0 +1,37 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+
+@Component({
+  selector: 'example-code-viewer',
+  templateUrl: './example-code-viewer.html',
+  styleUrls: ['./example-viewer.scss'],
```

Comment:
```suggestion
  styleUrls: ['./example-viewer.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
```

You're default change detection strategy should be `OnPush` for performance.